docs/docs/sample-scripts/spo/copy-librarys-folders-and-contents-to-another-library/index.md

```diff
@@ -0,0 +1,68 @@
+---
+tags:
+  - files
+  - folders
+  - libraries
+  - migration
+---
+
+# Copy library's folders and contents to another library
+
+Author: [Nico De Cleyre](https://www.nicodecleyre.com)
+
+The following script shows how you can copy all library's folders and contents to another library
+
+=== "PowerShell"
+
+    ```powershell
+    param
+    (
+        [Parameter(Mandatory = $true, HelpMessage = "The source site url")][string] $SiteURL,
+        [Parameter(Mandatory = $true, HelpMessage = "The server- or site-relative URL of the source document library")][string] $SourceFolder,
+        [Parameter(Mandatory = $true, HelpMessage = "The server- or site-relative URL of the destination document library")][string] $DestinationFolder,
+        [Parameter(Mandatory = $false, HelpMessage = "The destination site url")][string] $DestinationUrl
+    )
+    try {
+        #Connect to m365
+        $M365Status = m365 status
+        if ($M365Status -match "Logged Out") {
+            Write-Host "Logging in the User!"
+            m365 login --authType browser
+        }
+
+        # Set the destination URL the same as the source URL if it's not given
+        if($DestinationUrl -eq $null){
```

Comment:
this is a bit tricky as if we do not specify the `DestinationUrl` param it is empty string not `null` 
so when we leave it empty we may get an error 
![image](https://github.com/pnp/cli-microsoft365/assets/58668583/2af15bcf-0e67-4d06-a16b-9fcd07a66a39)

Unless we specify it or specify it as `$null` 🤔. 
We could update this condition to something like `... -eq '' )` or maybe we could play with adding a default value to this param